testdata/p4_14_samples_outputs/switch_20160512/switch-first.p4

```diff
@@ -2937,7 +2937,7 @@ control process_egress_acl(inout headers hdr, inout metadata meta, inout standar
     }
     @name(".egress_mirror") action egress_mirror(bit<32> session_id) {
         meta.i2e_metadata.mirror_session_id = (bit<16>)session_id;
-        clone3<tuple<bit<32>, bit<16>>>(CloneType.E2E, session_id, { meta.i2e_metadata.ingress_tstamp, meta.i2e_metadata.mirror_session_id });
+        clone3<tuple<bit<32>, bit<16>>>(CloneType.E2E, (bit<32>)(bit<16>)session_id, { meta.i2e_metadata.ingress_tstamp, meta.i2e_metadata.mirror_session_id });
```

Comment:
Turns out the problem is Dag vs tree in the visitor -- because this node has mulitple parents in the tree and those contexts have different types, it needs different casts in the different uses.  The fix is to set `visitDagOnce = false` in the `MakeImplicitCastsExplicit` transform, but this is a subtle issue that is easily missed.

Good catch!